Resource `azurerm_cognitive_account_rai_policy` requires `severity_threshold` where does not have an effect

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

While transitioning from `azapi_resource` to `azurerm_cognitive_account_rai_policy` I noticed the `severity_threshold` attribute in is required although it does not seem to have any effect for the following content filter types:

Jailbreak
Indirect Attack
Protected Material Text
Protected Material Code


## Azure OpenAI Service
Looking at the Azure OpenAi settings it does not seem possible to specify values for these filters:
![Image](https://github.com/user-attachments/assets/5a8ba3bc-7ccb-4705-9f57-e2fb3bac3e5f)
![Image](https://github.com/user-attachments/assets/eca6230c-4c0a-4041-a8c0-e993b7dd8b25)


<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.10.5

### AzureRM Provider Version

4.17.0

### Affected Resource(s)/Data Source(s)

azurerm_cognitive_account_rai_policy

### Terraform Configuration Files

```hcl
resource "azurerm_cognitive_account_rai_policy" "content_filter_policy" {
  content_filter {
    name               = "Jailbreak" # or "Indirect Attack", "Protected Material Text", "Protected Material Code"
    filter_enabled     = true
    block_enabled      = true
    # severity_threshold = "High" <- value required in provider but not used in azure
    source             = "Prompt"
  }
}
```

### Debug Output/Panic Output

```shell
2025-01-31T16:22:07.374+0100 [DEBUG] provider: plugin exited
2025-01-31T16:22:07.390+0100 [ERROR] vertex "module.azure_openai.azurerm_cognitive_account_rai_policy.content_filter_policy" error: Missing required argument
2025-01-31T16:22:07.398+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-01-31T16:22:07.398+0100 [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/hashicorp/kubernetes/2.35.1/darwin_arm64/terraform-provider-kubernetes_v2.35.1_x5 id=98966
2025-01-31T16:22:07.398+0100 [DEBUG] provider: plugin exited
╷
│ Error: Missing required argument
│ 
│   on ../modules/azure-openai/azure-openai.tf line 94, in resource "azurerm_cognitive_account_rai_policy" "content_filter_policy":
│   94:   content_filter {
│ 
│ The argument "severity_threshold" is required, but no definition was found.
╵
2025-01-31T16:22:07.448+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-01-31T16:22:07.449+0100 [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/hashicorp/azurerm/4.17.0/darwin_arm64/terraform-provider-azurerm_v4.17.0_x5 id=98968
2025-01-31T16:22:07.449+0100 [DEBUG] provider: plugin exited
```

### Expected Behaviour

`severity_threshold` is not required for `content_filter` with names:
- Jailbreak
- Indirect Attack
- Protected Material Text
- Protected Material Code

### Actual Behaviour

`severity_threshold` is required for all `content_filter` names

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

- https://github.com/hashicorp/terraform-provider-azurerm/issues/22822
- https://github.com/hashicorp/terraform-provider-azurerm/pull/28013

--- a/internal/services/cognitive/cognitive_account_rai_policy_resource.go
+++ b/internal/services/cognitive/cognitive_account_rai_policy_resource.go
@@ -78,23 +78,23 @@ func (r CognitiveAccountRaiPolicyResource) Arguments() map[string]*pluginsdk.Sch
 						Required:     true,
 						ValidateFunc: validation.StringIsNotEmpty,
 					},
-					"filter_enabled": {
-						Type:     pluginsdk.TypeBool,
-						Required: true,
-					},
 					"block_enabled": {
 						Type:     pluginsdk.TypeBool,
 						Required: true,
 					},
-					"severity_threshold": {
-						Type:         pluginsdk.TypeString,
-						Required:     true,
-						ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForContentLevel(), false),
+					"filter_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
 					},
 					"source": {
 						Type:         pluginsdk.TypeString,
 						Required:     true,
 						ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForRaiPolicyContentSource(), false),
+					},
+					"severity_threshold": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForContentLevel(), false),
 					},
 				},
 			},
@@ -311,13 +311,18 @@ func expandRaiPolicyContentFilters(filters []AccountRaiPolicyContentFilter) *[]r
 
 	contentFilters := make([]raipolicies.RaiPolicyContentFilter, 0, len(filters))
 	for _, filter := range filters {
-		contentFilters = append(contentFilters, raipolicies.RaiPolicyContentFilter{
-			Name:              pointer.To(filter.Name),
-			Enabled:           pointer.To(filter.FilterEnabled),
-			Blocking:          pointer.To(filter.BlockEnabled),
-			SeverityThreshold: pointer.To(raipolicies.ContentLevel(filter.SeverityThreshold)),
-			Source:            pointer.To(raipolicies.RaiPolicyContentSource(filter.Source)),
-		})
+		f := raipolicies.RaiPolicyContentFilter{
+			Name:     pointer.To(filter.Name),
+			Enabled:  pointer.To(filter.FilterEnabled),
+			Blocking: pointer.To(filter.BlockEnabled),
+			Source:   pointer.To(raipolicies.RaiPolicyContentSource(filter.Source)),
+		}
+
+		if filter.SeverityThreshold != "" {
+			f.SeverityThreshold = pointer.To(raipolicies.ContentLevel(filter.SeverityThreshold))
+		}
+
+		contentFilters = append(contentFilters, f)
 	}
 	return &contentFilters
 }

--- a/internal/services/cognitive/cognitive_account_rai_policy_resource.go
+++ b/internal/services/cognitive/cognitive_account_rai_policy_resource.go
@@ -19,6 +19,7 @@ import (
 )
 
 var _ sdk.ResourceWithUpdate = &CognitiveAccountRaiPolicyResource{}
+var _ sdk.ResourceWithCustomizeDiff = &CognitiveAccountRaiPolicyResource{}
 
 type CognitiveAccountRaiPolicyResource struct{}
 
@@ -43,6 +44,55 @@ type AccountRaiPolicyResourceModel struct {
 	ContentFilter  []AccountRaiPolicyContentFilter `tfschema:"content_filter"`
 	Mode           string                          `tfschema:"mode"`
 	Tags           map[string]string               `tfschema:"tags"`
+}
+
+var severityThresholdNotApplicableFilterNames = []string{
+	"Jailbreak",
+	"Indirect Attack",
+	"Protected Material Text",
+	"Protected Material Code",
+}
+
+func (r CognitiveAccountRaiPolicyResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			if metadata.ResourceDiff == nil {
+				return nil
+			}
+
+			rawFilters, ok := metadata.ResourceDiff.GetOk("content_filter")
+			if !ok {
+				return nil
+			}
+
+			filters, ok := rawFilters.([]interface{})
+			if !ok {
+				return nil
+			}
+
+			for i, rawFilter := range filters {
+				filter, ok := rawFilter.(map[string]interface{})
+				if !ok {
+					continue
+				}
+
+				name, _ := filter["name"].(string)
+				severityThreshold, _ := filter["severity_threshold"].(string)
+
+				if severityThreshold == "" {
+					continue
+				}
+
+				for _, notApplicable := range severityThresholdNotApplicableFilterNames {
+					if name == notApplicable {
+						return fmt.Errorf("`severity_threshold` is not applicable for `content_filter[%d]` with name %q", i, name)
+					}
+				}
+			}
+
+			return nil
+		},
+	}
 }
 
 func (r CognitiveAccountRaiPolicyResource) Arguments() map[string]*pluginsdk.Schema {

--- a/internal/services/cognitive/cognitive_account_rai_policy_resource.go
+++ b/internal/services/cognitive/cognitive_account_rai_policy_resource.go
@@ -18,8 +18,10 @@ import (
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
 )
 
-var _ sdk.ResourceWithUpdate = &CognitiveAccountRaiPolicyResource{}
-var _ sdk.ResourceWithCustomizeDiff = &CognitiveAccountRaiPolicyResource{}
+var (
+	_ sdk.ResourceWithUpdate        = &CognitiveAccountRaiPolicyResource{}
+	_ sdk.ResourceWithCustomizeDiff = &CognitiveAccountRaiPolicyResource{}
+)
 
 type CognitiveAccountRaiPolicyResource struct{}
 

--- a/internal/services/cognitive/cognitive_account_rai_policy_resource_test.go
+++ b/internal/services/cognitive/cognitive_account_rai_policy_resource_test.go
@@ -81,6 +81,21 @@ func TestCognitiveAccountRaiPolicy_update(t *testing.T) {
 	})
 }
 
+func TestCognitiveAccountRaiPolicy_withoutSeverityThreshold(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cognitive_account_rai_policy", "test")
+	r := RaiPolicyTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.withoutSeverityThreshold(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
 func (r RaiPolicyTestResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
 	id, err := raipolicies.ParseRaiPolicyID(state.ID)
 	if err != nil {
@@ -187,4 +202,46 @@ resource "azurerm_cognitive_account_rai_policy" "import" {
   }
 }
 `, template)
+}
+
+func (r RaiPolicyTestResource) withoutSeverityThreshold(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-cognitive-%d"
+  location = "%s"
+}
+
+resource "azurerm_cognitive_account" "test" {
+  name                = "acctestcogacc-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  kind                = "OpenAI"
+  sku_name            = "S0"
+}
+
+resource "azurerm_cognitive_account_rai_policy" "test" {
+  name                 = "acctestraip-%s"
+  cognitive_account_id = azurerm_cognitive_account.test.id
+  base_policy_name     = "Microsoft.Default"
+
+  content_filter {
+    name           = "Jailbreak"
+    filter_enabled = true
+    block_enabled  = true
+    source         = "Prompt"
+  }
+
+  content_filter {
+    name               = "Hate"
+    filter_enabled     = true
+    block_enabled      = true
+    severity_threshold = "High"
+    source             = "Prompt"
+  }
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomString)
 }

--- a/website/docs/r/cognitive_account_rai_policy.html.markdown
+++ b/website/docs/r/cognitive_account_rai_policy.html.markdown
@@ -66,13 +66,13 @@ A `content_filter` block supports the following:
 
 * `name` - (Required) The name of the content filter.
 
-* `filter_enabled` - (Required) Whether the filter is enabled. Possible values are `true` or `false`.
-
 * `block_enabled` - (Required) Whether the filter should block content. Possible values are `true` or `false`.
 
-* `severity_threshold` - (Required) The severity threshold for the filter. Possible values are `Low`, `Medium` or `High`.
+* `filter_enabled` - (Required) Whether the filter is enabled. Possible values are `true` or `false`.
 
 * `source` - (Required) Content source to apply the content filter. Possible values are `Prompt` or `Completion`.
+
+* `severity_threshold` - (Optional) The severity threshold for the filter. Possible values are `Low`, `Medium` or `High`. This is not applicable for filter types such as `Jailbreak`, `Indirect Attack`, `Protected Material Text`, and `Protected Material Code`.
 
 ## Attributes Reference
 

--- a/website/docs/r/cognitive_account_rai_policy.html.markdown
+++ b/website/docs/r/cognitive_account_rai_policy.html.markdown
@@ -72,7 +72,9 @@ A `content_filter` block supports the following:
 
 * `source` - (Required) Content source to apply the content filter. Possible values are `Prompt` or `Completion`.
 
-* `severity_threshold` - (Optional) The severity threshold for the filter. Possible values are `Low`, `Medium` or `High`. This is not applicable for filter types such as `Jailbreak`, `Indirect Attack`, `Protected Material Text`, and `Protected Material Code`.
+* `severity_threshold` - (Optional) The severity threshold for the filter. Possible values are `Low`, `Medium` or `High`.
+
+-> **Note:** This is not applicable for filter types such as `Jailbreak`, `Indirect Attack`, `Protected Material Text`, and `Protected Material Code`.
 
 ## Attributes Reference
 

